tests/test_filters.py

```diff
@@ -118,3 +120,13 @@ def test_capitalize_first_for_non_strings():
     assert capitalize_first([{'list': 'of'}, 'things']) == [{'list': 'of'}, 'Things']
     assert capitalize_first({'this': 'thing'}) == {'this': 'thing'}
     assert capitalize_first('https://www.example.com') == 'https://www.example.com'
+
+
+def test_preserve_line_breaks():
+    assert preserve_line_breaks(mock.Mock(autoescape=False), '\r\n') == '<br>'
+    assert preserve_line_breaks(mock.Mock(autoescape=False), '\r\n\r\n') == '<br><br>'
+    assert preserve_line_breaks(mock.Mock(autoescape=False), '\r\n \r\n \r\n') == '<br><br>'
+    assert preserve_line_breaks(mock.Mock(autoescape=False), '\r\n\r\n\r\n\r\n\r\n\r\n') == '<br><br>'
+    assert preserve_line_breaks(mock.Mock(autoescape=False), '') == ''
+    assert preserve_line_breaks(mock.Mock(autoescape=False), '\r\n<h2>') == '<br>&lt;h2&gt;'
+    assert preserve_line_breaks(mock.Mock(autoescape=False), 'You’ll be working') == 'You’ll be working'
```

Comment:
It would be good to check a couple of other things that won't work, like 
- `"\n"` (should be `"\n"`)
- `"\r\n\r\n  \r\n"`  (should be `"<br><br>  <br>"`)